src/System.Globalization/tests/CharNumericCategory/GetNumericValue.cs

```diff
@@ -0,0 +1,145 @@
+            TestStringInputWithIndex(String.Empty, "ending text", true);
+        }
+
+        [Fact]
+        public void InputStringBeginningOfStringWithoutValue()
+        {
+            TestStringInputWithIndex(String.Empty, "ending text", false);
+        }
+
+        [Fact]
+        public void VerifyIndexRange()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => CharUnicodeInfo.GetNumericValue("test", -1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => CharUnicodeInfo.GetNumericValue("test", 4));
+        }
+
+        /// <summary>
+        /// Iterates through all unicode data that can be cast to char and verifies that the Numeric 
+        /// value is as expected.
+        /// 
+        /// If the code point does not have a value, then the expected value from GetNumericValue
+        /// will be -1.
+        /// </summary>
+        /// <param name="hasValue">Verify Unicode codepoints that have a value or not</param>
+        private void TestCharInput(bool hasValue)
+        {
+            int count = 0;
+
+            foreach (var item in Factory.GetUnicodeData())
+            {
+                // Skip items that have a value
+                if (item.CodeValue < char.MaxValue && item.NumericValue.HasValue == hasValue)
+                {
+                    var actualNumericValue = CharUnicodeInfo.GetNumericValue(checked((char)item.CodeValue));
+
+                    Assert.Equal(hasValue ? item.NumericValue.Value : -1, actualNumericValue);
+
+                    count++;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Iterate through all values and convert each to a string.  Padding for the front or back can be used
+        /// to test the indexing argument.
+        /// 
+        /// If the code point does not have a value, then the expected value from GetNumericValue
+        /// will be -1.
+        /// </summary>
+        /// <param name="hasValue">Verify Unicode codepoints that have a value or not</param>
+        private void TestStringInputWithIndex(string frontPadding, string backPadding, bool hasValue)
+        {
+            int count = 0;
+
+            foreach (var item in Factory.GetUnicodeData())
+            {
+                // Skip surrogate pair range from U+D800 through U+DFFF
+                if (item.CodeValue >= 0xd800 && item.CodeValue <= 0xdfff)
+                    continue;
+
+                // Skip characters without a value
+                if (item.NumericValue.HasValue != hasValue)
+                    continue;
+
+                var unicodeStr = frontPadding + Char.ConvertFromUtf32(item.CodeValue) + backPadding;
+                var actualNumericValue = CharUnicodeInfo.GetNumericValue(unicodeStr, frontPadding.Length);
+
+                Assert.Equal(hasValue ? item.NumericValue.Value : -1, actualNumericValue);
+
+                count++;
+            }
+
+            Console.WriteLine("Tested {0} characters.", count);
```

Comment:
Yes, please remove the remaining uses of Console.WriteLine.... looks like there are only a few in the whole PR, so it should be easy.